var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -156,6 +156,20 @@ def configure_args(self):
 
         return ["--with-zlib=%s" % spec['zlib'].prefix] + extra_args
 
+    def configure(self, spec, prefix):
+        # Run the default autotools package configure
+        super(Hdf5, self).configure(spec, prefix)
+
+        if '@:1.8.14' in spec:
```

Comment:
I've been thinking about all of these things for so long! Here are my two cents:

```python
@AutotoolsPackage.precondition('configure')
@AutotoolsPackage.sanity_check('configure')
```
Meh, it's alright. It makes sense when they are actually preconditions and sanity checks, but a lot of the time they are writing `Make.inc` files or filtering out compilers.
```python
@AutotoolsPackage.run_before('configure')
@AutotoolsPackage.run_after('configure')
```
Much easier to understand.
```python
@run_before('configure')
@run_after('configure')
```
Less intimidating. No need to duplicate the base class name.
```python
def pre_configure(self):
def post_configure(self):
```
I think this solution is easiest for Python beginners to understand. What I'm hoping is that `package.py` files will be so basic that people with absolutely no Python experience can figure out how to create and edit them. Decorators are something that weren't taught in my intro Python course, and I think we should avoid them if we can.